client/src/components/signUpForm/BasicUserInfo/index.js

```diff
@@ -0,0 +1,201 @@
+
+import InputField from '../../InputField';
+import Container from '../../Container';
+import ButtonComponent from '../../Button';
+import schema, {
+  emailSchema,
+  firstNameSchema,
+  lastNameSchema,
+  passwordSchema,
+} from './validation';
+
+const { shape, func } = PropTypes;
+
+const useStyle = makeStyles((theme) => ({
+  input: {
+    color: theme.customColors[1],
+    width: '80%',
+    backgroundColor: theme.customColors[7],
+  },
+}));
+
+function BasicUserInfo(props) {
+  const {
+    methods: { handleNext, setData },
+  } = props;
+  const classes = useStyle();
+
+  const [firstName, setFirstName] = useState('');
+  const [lastName, setLastName] = useState('');
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
+
+  const [firstNameError, setFirstNameError] = useState(false);
+  const [lastNameError, setLastNameError] = useState(false);
+  const [emailError, setEmailError] = useState(false);
+  const [passwordError, setPasswordError] = useState(false);
+
+  return (
+    <Container screenSize="lg" direction="row" itemColumns="6" spacing={2}>
+      <Hidden smDown key="1">
+        <CardMedia
+          component="img"
+          height="600"
+          image="./signupImages/3.png"
+          title="main"
+        />
+      </Hidden>
+
+      <Container
+        key="2"
+        direction="column"
+        itemColumns="12"
+        spacing={5}
+        style={{
+          width: '100%',
+          border: '1px solid blue',
+          borderRadius: '12px',
+          padding: '15px 0',
+        }}
+      >
+        <Typography key="1" variant="h4">
+          Create User Account :-
+        </Typography>
+        <form
+          key="2"
+          style={{
+            width: '100%',
+          }}
+        >
+          <Container direction="column" itemColumns="12" spacing={5}>
+            <InputField
+              key="1"
+              variant="outlined"
+              label="First Name"
+              className={classes.input}
+              type="text"
+              onChange={async ({ target: { value } }) => {
+                setFirstName(value);
+                const isValid = await firstNameSchema.isValid({ firstName });
+                setFirstNameError(() => !isValid);
+              }}
+              value={firstName}
+              placeholder="Enter First Name ..."
+              error={!!firstNameError}
+              helperText={
+                firstNameError ? 'Please Enter A valid First Name' : null
+              }
+            />
+            <InputField
+              key="2"
+              variant="outlined"
+              label="Last Name"
+              className={classes.input}
+              type="text"
+              onChange={async ({ target: { value } }) => {
+                setLastName(value);
+                const isValid = await lastNameSchema.isValid({ lastName });
+                setLastNameError(() => !isValid);
+              }}
```

Comment:
This onChange validation pattern is repeated several times in this component so there might be a way to write a helper function to remove some of the duplication.  At the top of the file, outside the component, you could have a helper function like this:
```
const updateAndValidateInput = (schemaKey, schema, setValue, setIsValid) => {
 return async ({ target: {value } }) => {
    setValue(value);
    const isValid = await schema.isValid({[schemaKey]: value});
    setIsValid(isValid);
 };
});
```
Notice that this function is returning another function, and that returned function is what'll be passed to onChange and it takes an event as an argument.  So for example, you'd use it like this:
```
onChange={updateAndValidateInput('lastName', lastNameSchema, setLastName, setLastNameError)}
```
You don't have to do this (and my code hasn't been tested and might not work :)) but it's something you can keep in mind for next time.  It's trading more complicated code for less duplication, which might be good overall because the problem with duplication comes when you have to maintain or make a change in several places instead of just one place.